src/Engine.php

```diff
@@ -0,0 +1,201 @@
+            $searchableEntities = [$searchableEntities];
+        }
+
+        $data = [];
+        foreach ($searchableEntities as $entity) {
+            $searchableArray = $entity->getSearchableArray();
+            if ($searchableArray === null || count($searchableArray) === 0) {
+                continue;
+            }
+
+            $indexName = $entity->getIndexName();
+
+            if (!isset($data[$indexName])) {
+                $data[$indexName] = [];
+            }
+
+            $data[$indexName][] = $searchableArray + [
+                    'objectID' => $entity->getId(),
+                ];
+        }
+
+        $result = [];
+        foreach ($data as $indexName => $objects) {
+            if (!array_key_exists('autoGenerateObjectIDIfNotExist', $requestOptions)) {
+                $requestOptions['autoGenerateObjectIDIfNotExist'] = true;
+            }
+
+            $result[$indexName] = $this->client
+                ->initIndex($indexName)
+                ->saveObjects($objects, $requestOptions);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Remove entities from Algolia.
+     *
+     * @param array<int, SearchableEntity>    $searchableEntities
+     * @param array<string, int|string|array> $requestOptions
+     *
+     * @return array<string, BatchIndexingResponse>
+     *
+     * @throws \Algolia\AlgoliaSearch\Exceptions\AlgoliaException
+     */
+    public function remove($searchableEntities, $requestOptions = [])
+    {
+        if ($searchableEntities instanceof SearchableEntity) {
+            $searchableEntities = [$searchableEntities];
+        }
+
+        $data = [];
+        foreach ($searchableEntities as $entity) {
+            $searchableArray = $entity->getSearchableArray();
+            if ($searchableArray === null || count($searchableArray) === 0) {
+                continue;
+            }
+            $indexName = $entity->getIndexName();
+
+            if (!isset($data[$indexName])) {
+                $data[$indexName] = [];
+            }
+
+            $data[$indexName][] = $entity->getId();
+        }
+
+        $result = [];
+        foreach ($data as $indexName => $objects) {
+            $result[$indexName] = $this->client
+                ->initIndex($indexName)
+                ->deleteObjects($objects, $requestOptions);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Clear all objects from the index.
+     *
+     * @param string $indexName
+     *
+     * @return \Algolia\AlgoliaSearch\Response\AbstractResponse
+     */
+    public function clear($indexName)
+    {
+        $index = $this->client->initIndex($indexName);
+
+        if ($index->exists()) {
+            return $index->clearObjects();
+        }
+
+        return new NullResponse();
+    }
+
+    /**
+     * Delete the index.
+     *
+     * @param string $indexName
+     *
+     * @return \Algolia\AlgoliaSearch\Response\AbstractResponse
+     */
+    public function delete($indexName)
```

Comment:
No $requestOptions here?

Reply:
Very nice catch. @chloelbn let's make sure the request options are applied cross all public methods of the searchService.